### PR TITLE
feat: SITES-20226 Localize Generate Variations app

### PR DESCRIPTION
**Pull request by askayastha22**
Demo URL: https://experience.adobe.com/?shell_source=dev#/@sitesinternal/aem/generate-variations
## Description

To see the app localized in Tier 1 languages, please go to Unified Shell's [preferences](https://experience.adobe.com/#/@sitesinternal/preferences/) and change the language in your profile.
## Related Issue
https://jira.corp.adobe.com/browse/SITES-20226

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: web-src/src/components/App.js [hunk 1]
```diff
@@ -16,13 +16,26 @@ import {
...
 import { MainSidePanel } from './MainSidePanel.js';
 import { PromptSessionPanel } from './PromptSessionPanel.js';
 import { PromptTemplateLibraryPanel } from './PromptTemplateLibraryPanel.js';
 import { viewTypeState, ViewType } from '../state/ViewType.js';
 import { mainSidePanelState, MainSidePanelType } from '../state/MainSidePanelState.js';
 import { FavoriteVariantListPanel } from './FavoriteVariantListPanel.js';
+import { useShellContext } from './ShellProvider.js';
+
+/* eslint-disable import/extensions */
+import * as messages1 from './__localization__/App.l10n';
```
Comment (basecode) [inline web-src/src/components/App.js:32]: Can this be renamed to `messagesApp` and `messagesMainSidePanel` etc?

File: web-src/src/components/__localization__/PromptSessionSideView.l10n/de_de.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "promptSessionSideView.cancelActionLabel": "Abbrechen",
+  "promptSessionSideView.deleteActionLabel": "Löschen",
+  "promptSessionSideView.editPromptButtonLabel": "Eingabeaufforderung bearbeiten",
+  "promptSessionSideView.generateButtonContextualHelpContent": "<p>Der Zugriff auf diese Funktion unterliegt Ihrer Zustimmung zu {legalTermsLink}und Folgendes:</p><ul><li>Aufforderungen, Kontext oder zusätzliche Informationen oder andere Eingaben, die Sie zu dieser Funktion (a) bereitstellen, müssen an einen bestimmten Kontext gebunden sein, der Ihr Branding-Material, Ihren Website-Inhalt, Daten, Schemas für diese Daten, Vorlagen oder andere vertrauenswürdige Dokumente umfassen kann, und (b) dürfen keine persönlichen Informationen enthalten (persönliche Informationen umfassen alles, das zu einer bestimmten Person zurückverlinkt werden kann).</li><li>Sie sollten alle Ausgaben aus dieser Funktion auf Genauigkeit überprüfen und sicherstellen, dass sie für Ihren Anwendungsfall geeignet sind.</li></ul>",
+  "promptSessionSideView.generateButtonContextualHelpHeading": "Nutzungsbedingungen",
+  "promptSessionSideView.generateButtonLabel": "Erzeugen",
```
Comment (basecode) [inline web-src/src/components/__localization__/PromptSessionSideView.l10n/de_de.json:10]: "Generieren"

**Review by basecode — APPROVED** [review]

File: web-src/src/components/App.l10n.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    cancelActionLabel: {
+      defaultMessage: 'Cancel',
+      id: 'app.cancelActionLabel',
+      description: 'Label for Cancel action',
+    },
+    consentDialogHeading: {
+      defaultMessage: 'Generative AI in Adobe apps',
+      id: 'app.consentDialogHeading',
+      description: 'Heading for consent dialog',
```
Comment (vtsaplin) [inline web-src/src/components/App.l10n.js:73]: Where do the descriptions come from?

File: web-src/src/components/App.js (hunk 1 above)
Comment (vtsaplin) [reply web-src/src/components/App.js:32]: We might simplify message loading with a webpack plugin, though it's not urgent. We can refactor this later.

**Review by vtsaplin — APPROVED** [review]

**Pull request merged**